test/functional/tool_bitcoin_chainstate.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+import subprocess
+
+from test_framework.test_framework import BitcoinTestFramework
+
+class BitcoinChainstateTest(BitcoinTestFramework):
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_bitcoin_chainstate()
+
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.chain = ""
+        self.num_nodes = 1
+
+    def add_block(self, datadir, input, expected_stderr):
+        proc = subprocess.Popen(
+            self.get_binaries().chainstate_argv() + [datadir],
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True
+        )
+        stdout, stderr = proc.communicate(input=input, timeout=5)
+        self.log.debug("STDOUT: {0}".format(stdout.strip('\n')))
+        self.log.info("STDERR: {0}".format(stderr.strip('\n')))
+
+        if expected_stderr not in stderr:
+            raise AssertionError(f'Expected stderr output {expected_stderr} does not partially match stderr:\n{stderr}')
+
+    def run_test(self):
+        node = self.nodes[0]
+        datadir = node.cli.datadir
+        # This might trigger a warning that there is not enough disk space, since this node runs on mainnet. Ignore it.
```

Comment:
FWIW #32057 went with `-prune=550` to avoid this condition, i don't know if that's possible here but would prefer to be consistent.